src/ViewListingDisplay/components/Listing.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+import PostCommentsDisplay from '../../PostCommentsDisplay/components/PostCommentsDisplay.jsx';
+
+import '../styles/Listing.css';
+
+/** Functional Component that creates a listing. Takes in listingInfo as a prop 
+(which holds the title, description, listingPrice, etc.) to fill in the listing */
+function Listing(props) {
+    const { 
+        title, 
+        description, 
+        listingPrice, 
+        numRooms, 
+        numSingles, 
+        numShared, 
+        singlePrice, 
+        sharedPrice, 
+        leaseType, 
+        startDate, 
+        endDate, 
+        numBathrooms 
+    } = props.listingInfo;
+
+    const roomInfo = getRoominfo(numSingles, numShared, singlePrice, sharedPrice);
+    const leaseTypeText = (leaseType === "YEAR_LONG") ? "Yearly" : "Monthly";
+    return (
+        <div className="listing-box">
+            <h1 className="title">{title}</h1>
+            <p className="description">{description}</p>
+            <p className="listingPrice">Listing Price: {listingPrice}</p>
+            <p className="leaseType">Lease Type: {leaseTypeText} Lease</p>
+            <div className="avaiability-container">
+                <h3>Availability</h3>
+                <p>Start: {startDate}</p>
+                <p>End: {endDate}</p>
+            </div>
+            <div className="rooms-info-containter">
+                <h3>Room(s) Information:</h3>
+                <p>Total number of room(s): {numRooms}</p>
+                { roomInfo }
+                <p>Total number of bathroom(s): {numBathrooms}</p>
+            </div>
+            <PostCommentsDisplay/>
+        </div>
+    );
+}
+
+function getRoominfo(numSingles, numShared, singlePrice, sharedPrice){
+    if((numSingles === '0' || numShared === 'USD 0') && (singlePrice === '0' || sharedPrice === 'USD 0')){
```

Comment:
do we need USD for numShared? Isn't it number of shared room?
What about singlePrice? May be we need USD there

This condition is little odd too. 
If numSingles is '0' , singlePrice should be 'USD 0' too. And even though we have numShared, we will return "No rooms available". Is this what we want?

Reply:
Yeah I forgot to commit the change that fixed this. if should be 
if((numSingles === '0' || singlePrice === 'USD 0') && (numShared === '0' || sharedPrice === 'USD 0')){

Fixed